### PR TITLE
tmux 2.2

### PR DESCRIPTION
**Pull request by seryl**
- [√] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [√] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [√] Does your submission pass `brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [√] Have you built your formula locally prior to submission with `brew install <formula>`?

### CODE REVIEW

File: Formula/tmux.rb [hunk 1]
```diff
@@ -50,12 +41,7 @@ def install
 
     system "make", "install"
 
-    if build.head?
-      pkgshare.install "example_tmux.conf"
-    else
-      bash_completion.install "examples/bash_completion_tmux.sh" => "tmux"
-      pkgshare.install "examples"
-    end
```
Comment (alexpearce) [inline Formula/tmux.rb]: The `example_tmux.conf` file [still exists](https://github.com/tmux/tmux/blob/9a4b45dc0fdefead3fe9d471e144da78163860d0/example_tmux.conf), so the `pkgshare.install "example_tmux.conf"`line can be kept.

File: Formula/tmux.rb [hunk 2]
```diff
@@ -3,22 +3,8 @@ class Tmux < Formula
...
-
-    patch do
-      # This fixes the Tmux 2.1 update that breaks "tmux killw\; detach"
-      # https://github.com/tmux/tmux/issues/153#issuecomment-150184957
-      url "https://github.com/tmux/tmux/commit/3ebcf25149d75977ea97e9d4f786e0508d1a0d5e.diff"
-      sha256 "65a8bc0b2f6a8b41ad27605fd99419fff36314499969adc9d17dd3940a173508"
-    end
+    url "https://github.com/tmux/tmux/releases/download/2.2/tmux-2.2.tar.gz"
+    sha256 "bc28541b64f99929fe8e3ae7a02291263f3c97730781201824c0f05d7c8e19e4"
   end
```
Comment (alexpearce) [inline Formula/tmux.rb]: I don't know what the Homebrew guidelines say about it, but the block is no longer necessary, so can just have the `url` and `sha256` methods:

```ruby
class Tmux < Formula
  desc "Terminal multiplexer"
  homepage "https://tmux.github.io/"
  url "https://github.com/tmux/tmux/releases/download/2.2/tmux-2.2.tar.gz"
  sha256 "bc28541b64f99929fe8e3ae7a02291263f3c97730781201824c0f05d7c8e19e4"
  # ...
end
```

**Comment by deiwin** [comment]
Please also consider including a patch for this issue: https://github.com/tmux/tmux/issues/328. There's a bug in OS X which causes some of the characters to not be displayed with tmux 2.2.

**Comment by mikemcquaid** [comment]
@deiwin Can you please submit this patch to the upstream developers of this project and add a link to the upstream patch submission and explanation of why the patch is needed in a comment that @seryl can add to the formula file. Thanks! I'd probably also like to see some indication that upstream will merge this.

**Comment by seryl** [comment]
It seems to resolve issues printing UTF8 characters inside of tmux on osx.

I'm getting it added now.

**Comment by deiwin** [comment]
@mikemcquaid, it's unlikely that it will be merged into the tmux project. From tmux's point of view this is an issue with OS X, not tmux itself, and the patch is simply a workaround for this.

**Comment by mikemcquaid** [comment]
@deiwin We don't accept patches that won't be merged upstream.

**Comment by seryl** [comment]
Pushed the asked updates without the UTF8 fix... It seems like we might want to wait until there's support for it.

**Comment by seryl** [comment]
Note: It doesn't appear to be all UTF8 chars; but there are likely a few.

https://github.com/tmux/tmux/issues/328

File: Formula/tmux.rb [hunk 3]
```diff
@@ -40,6 +23,11 @@ class Tmux < Formula
   depends_on "pkg-config" => :build
   depends_on "libevent"
 
+  resource "completion" do
+    url "https://raw.githubusercontent.com/przepompownia/tmux-bash-completion/678a27616b70c649c6701cae9cd8c92b58cc051b/completions/tmux"
+    sha256 "a0905c595fec7f0258fba5466315d42d67eca3bd2d3b12f4af8936d7f168b6c6"
+  end
```
Comment (mikemcquaid) [inline Formula/tmux.rb]: Would probably rather not install completion that's not "official" here.

Comment (mikemcquaid) [inline Formula/tmux.rb]: Again, I'd suggest trying to get this included upstream would be best.

**Comment by mikemcquaid** [comment]
Probably going to hold off merging this until there's more activity on that UTF-8 issue.

File: Formula/tmux.rb (hunk 3 above)
Comment (seryl) [inline Formula/tmux.rb]: Are we sure we don't want to? It's what is referenced under: https://github.com/tmux/tmux/blob/master/README#L48

Comment (mikemcquaid) [inline Formula/tmux.rb]: Oh, seems fine then :+1:. Could you ask them for a release tag and use the tarball from that? Thanks!

**Comment by apjanke** [comment]
No activity on the UTF-8 issue, and it's been 3 weeks.

Let's take a pass on 2.2 and see if it's fixed in a future point release.

**Comment by hdra** [comment]
FWIW, I installed from HEAD yesterday, and `printf '\u2b06\n'` prints the arrow head correctly.

`tmux -V` outputs `tmux 2.3`

**Comment by apjanke** [comment]
Great. In that case, I think we should just skip 2.2 in Homebrew and wait until the next one comes out.

Any objections?

**Comment by gkostyanikov** [comment]
https://github.com/tmux/tmux/issues/389 i'm still have this issue with russian input

**Comment by mikemcquaid** [comment]
We'll wait for 2.3.

**Comment by joshuarubin** [comment]
FWIW, the patch to fix the utf8 issue is pretty simple (tmux/tmux@23fdbc9ea6a6f5c93f042043f0407ed5d9bd0e5b). I created a working tmux 2.2 formula here: https://gist.github.com/joshuarubin/af7ed76c46ec5a9406a8621deb582614

**Comment by MikeMcQuaid** [comment]
@joshuarubin If that can go as https://github.com/tmux/tmux/commit/23fdbc.patch into a `patch do` block we'll accept a PR for it.

**Pull request closed without merging**